File: python/compatibility.py
# AstroAPI — Love compatibility between two people

import json
import os
import sys

import requests

API_KEY = os.environ.get("ASTROAPI_KEY")
BASE_URL = os.environ.get("ASTROAPI_BASE_URL", "https://api.astroapi.cloud")

if not API_KEY:
    print("Error: Set ASTROAPI_KEY environment variable first.", file=sys.stderr)
    print('  export ASTROAPI_KEY="your-api-key"', file=sys.stderr)
    sys.exit(1)

HEADERS = {
    "X-Api-Key": API_KEY,
    "Content-Type": "application/json"
}

payload = {
    "dateTime1": "1990-06-15T14:30",
    "location1": {
        "longitude": 4.89,
        "latitude": 52.37,
        "timezone": "Europe/Amsterdam"
    },
    "dateTime2": "1988-03-22T09:15",
    "location2": {
        "longitude": 2.35,
        "latitude": 48.86,
        "timezone": "Europe/Paris"
    },
    "language": "en",
    "includeText": True,
    "includeReadableEntities": True
}

response = requests.post(f"{BASE_URL}/api/calc/compatibility", headers=HEADERS, json=payload)
response.raise_for_status()

data = response.json()
print(json.dumps(data, indent=2))

File: javascript/compatibility.mjs
// Love compatibility between two people

const API_KEY = process.env.ASTROAPI_KEY;
const BASE_URL = process.env.ASTROAPI_BASE_URL || "https://api.astroapi.cloud";

if (!API_KEY) {
    console.error("Error: Set ASTROAPI_KEY environment variable first.");
    process.exit(1);
}

const response = await fetch(`${BASE_URL}/api/calc/compatibility`, {
    method: "POST",
    headers: {
        "X-Api-Key": API_KEY,
        "Content-Type": "application/json"
    },
    body: JSON.stringify({
        dateTime1: "1990-06-15T14:30",
        location1: {
            longitude: 4.89,
            latitude: 52.37,
            timezone: "Europe/Amsterdam"
        },
        dateTime2: "1988-03-22T09:15",
        location2: {
            longitude: 2.35,
            latitude: 48.86,
            timezone: "Europe/Paris"
        },
        language: "en",
        includeText: true,
        includeReadableEntities: true
    })
});

if (!response.ok) {
    console.error(`Error: ${response.status} ${response.statusText}`);
    console.error(await response.text());
    process.exit(1);
}

const data = await response.json();
console.log(JSON.stringify(data, null, 2));
